icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java

```diff
@@ -1141,4 +1150,180 @@ public void TestUResourceBundleCoverage() {
+
+        // Override data files (after using the API to update ICUBinary.OVERRIDE_FILES)
+
+        // Copy test override file to a temporary directory, set the override data files path.
+        File tmpDir = copyJavaRescToTmpDir("/com/ibm/icu/dev/data/override/test01/pt.res");
+        String path = tmpDir.toString();
+        ICUBinary.OVERRIDE_DATA_FILES.setDataPath(path);
+        assertRBStringEquals(localeID, "Override pattern for compact short pattern for 1000", "0G", testResourceKeyPath);
+        tmpDir.deleteOnExit();
+
+        // Reset override data files to empty
+
+        ICUBinary.OVERRIDE_DATA_FILES.setDataPath("");
+        assertRBStringEquals(localeID, "Normal (no override) pattern for compact short pattern for 1000", "0 mil", testResourceKeyPath);
+    }
+
+    /*
+     * Test that the parent locale chain is logically correct when there exist an override data file.
+     */
+    @Test
+    public void TestOverrideDataParentLocaleSingleFile() {
+        String testResourceKeyPath = "calendar/gregorian/availableFormats/yMd";
+
+        // Normal data files
+
+        // localeID, expStrVal
+        Object[][] normalDataFileCaseData = {
+                {"en_CA", "y-MM-dd"},
+                {"en",    "M/d/y"},
+                {"root",  "y-MM-dd"},
+        };
+        for (Object[] datum : normalDataFileCaseData) {
+            String localeID = (String) datum[0];
+            String expectedStrVal = (String) datum[1];
+
+            String assertMsg = "Gregorian yMd for " + localeID;
+
+            assertRBStringEquals(localeID, assertMsg, expectedStrVal, testResourceKeyPath);
+        }
+
+        // Override files - locale set = #{en_CA}
+
+        // Copy test override file to a temporary directory, set the override data files path.
+        File tmpDir = copyJavaRescToTmpDir("/com/ibm/icu/dev/data/override/test02/en_CA.res");
+        String path = tmpDir.toString();
+        ICUBinary.OVERRIDE_DATA_FILES.setDataPath(path);
+        tmpDir.deleteOnExit();
+
+        // localeID, expStrVal
+        Object[][] rootOverrideCaseData = {
+                {"en_CA", "test override pattern for en_CA"},
+                {"en",    "M/d/y"},
+                {"root",  "y-MM-dd"},
+        };
+        for (Object[] datum : rootOverrideCaseData) {
+            String localeID = (String) datum[0];
+            String expectedStrVal = (String) datum[1];
+
+            String assertMsg = "Gregorian yMd for " + localeID;
+
+            assertRBStringEquals(localeID, assertMsg, expectedStrVal, testResourceKeyPath);
+        }
+
+        // Reset override data files to empty
+
+        ICUBinary.OVERRIDE_DATA_FILES.setDataPath("");
+
+    }
+
+    /*
+     * Test that the parent locale chain is logically correct when there exist multiple override data files.
+     */
+    @Test
+    public void TestOverrideDataParentLocaleMultipleFiles() {
+        // Override files - locale set = #{en_CA, en, root}
```

Comment:
remove, or sync with the actual locale IDs (which changed)

Reply:
Done.